CsvParser/csvParser.go

```diff
@@ -129,3 +143,140 @@ func makeJsonObjFile(arr []incident.Incident) *os.File {
 
 	return f
 }
+
+//convert a default incident that implements our interface to a less safe incident type to be dumped on .json file
+func convertDefaultIncidentToIncidentData(i incident.DefaultIncident) incident.IncidentData {
+	gpStart, gpEnd := i.GetGoodPeriod()
+	bpStart, bpEnd := i.GetBadPeriod()
+	gMetric := i.GetGoodMetric()
+	bMetric := i.GetBadMetric()
+	severity := i.GetSeverity()
+	testsAffected := i.GetTestsAffected()
+	gpInfo := i.GetGoodPeriodInfo()
+	bpInfo := i.GetBadPeriodInfo()
+	iInfo := i.GetIncidentInfo()
+	locationString := i.GetLocation()
+	ASN := i.GetASN()
+	inc := incident.IncidentData{gpStart, gpEnd, bpStart, bpEnd, gMetric, bMetric, ASN, locationString, severity, testsAffected, gpInfo, bpInfo, iInfo}
+	return inc
+}
+
+// break the location code into approapriate locations
+func breakTheLocCodeDown(locationCode string) []string {
+	locationCodesArr := make([]string, 0)
+
+	if (len(locationCode)) > 6 {
+		for i := 0; i < 6; i = i + 2 {
+			locationCodesArr = append(locationCodesArr, locationCode[i:i+2])
+		}
+
+		locationCodesArr = append(locationCodesArr, locationCode[6:])
+
+		return locationCodesArr
+	}
+
+	for i := 0; i < len(locationCode); i = i + 2 {
+		locationCodesArr = append(locationCodesArr, locationCode[i:i+2])
+	}
+
+	return locationCodesArr
+}
+
+//check if na or eu locations directories exist for example
+func doesDirExist(originPath string, dir string) bool {
+	actualPath := originPath + "/" + dir
+
+	if _, err := os.Stat(actualPath); !os.IsNotExist(err) {
+		return true
+	}
+
+	return false
+}
+
+//check if a .json file of incidents for an asn(isp) exists in a specific location/dir
+func doesJsonFileExist(path string, asn string) (bool, string) {
+	filePath := path + "/" + asn + ".json"
+	info, err := os.Stat(filePath)
+
+	if os.IsNotExist(err) || info.IsDir() {
+		return false, filePath
+	}
+
+	return true, filePath
+}
+
+//construct location dir hierachy as you look at an incident and return the path that the incident ends up in
+func dynamicallyMakeDir(originPath string, locationCode string, asn string) string {
+	locationCodeArr := breakTheLocCodeDown(locationCode)
+	locationCodeArrLen := len(locationCodeArr)
+
+	for i := 0; i < locationCodeArrLen; i++ {
+
+		if !doesDirExist(originPath, locationCodeArr[i]) {
+			err := os.MkdirAll(originPath+"/"+locationCodeArr[i], 0755)
+
+			if err != nil {
+				log.Fatal(err)
+				//think about the implication of this error
+			}
+		}
+
+		originPath = originPath + "/" + locationCodeArr[i]
+	}
+
+	return originPath
+}
+
+func readJsonFileAddToIt(filenamepath string, i incident.DefaultIncident) {
```

Comment:
Change name to "appendToJson"

Reply:
instead of doesDirExist or doesJsonFileExist just name your functions dirExists and jsonFileExists.... for example if you had to use check if it exists you can say like
if dirExists: blahblahblah
if jsonFileExists: blahblahblah

Reply:
instead of doesDirExist or doesJsonFileExist just name your functions dirExists and jsonFileExists.... for example if you had to use check if it exists you can say like
if dirExists: blahblahblah
if jsonFileExists: blahblahblah